server/api/impl/store/manifest.go

```diff
@@ -0,0 +1,304 @@
+	s.mutex.Unlock()
+
+	return nil
+}
+
+func (s *manifestStore) Get(appID apps.AppID) (*apps.Manifest, error) {
+	s.mutex.RLock()
+	builtin := s.builtin
+	local := s.local
+	global := s.global
+	s.mutex.RUnlock()
+
+	m, ok := builtin[appID]
+	if ok {
+		return m, nil
+	}
+	m, ok = local[appID]
+	if ok {
+		return m, nil
+	}
+	m, ok = global[appID]
+	if ok {
+		return m, nil
+	}
+	return nil, utils.ErrNotFound
+}
+
+func (s *manifestStore) AsMap() map[apps.AppID]*apps.Manifest {
+	s.mutex.RLock()
+	builtin := s.builtin
+	local := s.local
+	global := s.global
+	s.mutex.RUnlock()
+
+	out := map[apps.AppID]*apps.Manifest{}
+	for id, m := range global {
+		out[id] = m
+	}
+	for id, m := range local {
+		out[id] = m
+	}
+	for id, m := range builtin {
+		out[id] = m
+	}
+	return out
+}
+
+func (s *manifestStore) StoreLocal(m *apps.Manifest) error {
+	conf := s.conf.GetConfig()
+	prevSHA := conf.LocalManifests[string(m.AppID)]
+
+	data, err := json.Marshal(m)
+	if err != nil {
+		return err
+	}
+	sha := fmt.Sprintf("%x", sha1.Sum(data)) // nolint:gosec
+	if sha == prevSHA {
+		return nil
+	}
+
+	_, err = s.mm.KV.Set(api.PrefixLocalManifest+sha, m)
+	if err != nil {
+		return err
+	}
+
+	s.mutex.RLock()
+	local := s.local
+	s.mutex.RUnlock()
+	updatedLocal := map[apps.AppID]*apps.Manifest{}
+	for k, v := range local {
+		if k != m.AppID {
+			updatedLocal[k] = v
+		}
+	}
+	updatedLocal[m.AppID] = m
+	s.mutex.Lock()
+	s.local = updatedLocal
+	s.mutex.Unlock()
+
+	updated := map[string]string{}
+	for k, v := range conf.LocalManifests {
+		updated[k] = v
+	}
+	updated[string(m.AppID)] = sha
+	sc := *conf.StoredConfig
+	sc.LocalManifests = updated
+	err = s.conf.StoreConfig(&sc)
+	if err != nil {
+		return err
+	}
+
+	_ = s.mm.KV.Delete(api.PrefixLocalManifest + prevSHA)
+	return nil
+}
+
+func (s *manifestStore) DeleteLocal(appID apps.AppID) error {
+	conf := s.conf.GetConfig()
+	sha := conf.LocalManifests[string(appID)]
+
+	err := s.mm.KV.Delete(api.PrefixLocalManifest + sha)
+	if err != nil {
+		return err
+	}
+
+	s.mutex.RLock()
+	local := s.local
+	s.mutex.RUnlock()
+	updatedLocal := map[apps.AppID]*apps.Manifest{}
+	for k, v := range local {
+		if k != appID {
+			updatedLocal[k] = v
+		}
+	}
+	s.mutex.Lock()
+	s.local = updatedLocal
+	s.mutex.Unlock()
+
+	updated := map[string]string{}
+	for k, v := range conf.LocalManifests {
+		updated[k] = v
+	}
+	delete(updated, string(appID))
+	sc := *conf.StoredConfig
+	sc.LocalManifests = updated
+
+	return s.conf.StoreConfig(&sc)
+}
+
+// GetManifest returns a manifest file for an app from the S3
+func (s *manifestStore) getFromS3(awscli awsclient.Client, bucket string, appID apps.AppID, version apps.AppVersion) ([]byte, error) {
+	name := fmt.Sprintf("manifest_%s_%s", appID, version)
+	data, err := awscli.GetS3(bucket, name)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to download manifest %s", name)
+	}
+	return data, nil
+}
+
+func validateManifest(m *apps.Manifest) error {
```

Comment:
That functionality does seem store specific. Should we move it next to the `Manifest` struct to be used in e.g the install command?